### ISSUE TITLE
Ability to set a weekly schedule with specific days when Light and Dark Auto Switches.

### ISSUE THREAD

**Issue by 16BitRun** [AutoDarkMode/Windows-Auto-Night-Mode#866]
Hi,

Is it possible to set a weekly schedule of when light mode with be on and when Dark Mode based on specific days on the week. For example, I work from home and I would like to automatically set Light Mode on my weekly Microsoft Teams Video Conferences on Mondays, Wednesdays and Fridays only from 7:30am - 2:30pm and then have it revert to Dark Mode. I know this is possible using Windows Task Manager but it would be great if this could be implemented into the Auto  Dark Mode app.

**Comment by Spiritreader** [comment]
We've been thinking about more sophisticated configuration for the timed events. Unfortunately it's quite a bit of work and there's no real time for it as ADM is a passion project.

Implementing this would require a substantial rewrite of how the dark and light states are calculated including changes to the postpone logic ("pause until next sunset/sunrise etc would need to be adjusted, wordings changed etc)

I would say that at this point in time it's unfortunately not very realistic for me (@Armin2208's role nowadays has more moved towards administration, testing and keeping the lights on for our store listing and ability to sign builds).

**Comment by Spiritreader** [comment]
@ChenYiLins this probably doesn't really fall into the "good first issue" category, this is more like "you know everything inside out about how theme switch governors work, and are very proficient in both frontend and backend c# programming" territory 😆

**Comment by ChenYiLins** [comment]
> [@ChenYiLins](https://github.com/ChenYiLins) this probably doesn't really fall into the "good first issue" category, this is more like "you know everything inside out about how theme switch governors work, 和 are very proficient in both frontend and backend c# programming" territory 😆

Yes, I'm sorry. Haha, this is really not an easy function to develop.

**Comment by Tomclanc** [comment]
I'd like to add a specific use case to this scheduling request: an optional lunch-break theme override that coexists with the usual sunrise/sunset schedule.

For example, use the normal solar schedule throughout the day, but automatically use dark mode from 12:20 to 13:30 during lunch, then resume whichever theme the normal schedule calls for at 13:30.

Suggested UI on the Time page:
- Disabled by default.
- Configurable start/end time and dark/light theme.
- Optional weekdays-only setting.
- Resume the underlying schedule after the interval; avoid redundant switches when both schedules agree.

I have made a small, unofficial personal prototype against 11.1.1.1: https://github.com/Tomclanc/AutoDarkMode-Personal-Patches . It reflects my personal aesthetic/use preferences, and the README explicitly recommends the original upstream app. It is offered only as a concrete example, not as a production-ready replacement for the scheduling work described above.

The prototype's 17 deterministic checks pass, including interval boundaries, weekday exclusions, overlapping sunset and configuration compatibility. Enabling a lunch interval also triggered a real dark-theme switch on my machine. End-of-interval restoration has been checked in calculation tests, not yet observed at the actual wall-clock boundary. I understand the concerns about postpone semantics and broader governor interactions; these would need full upstream review rather than treating this narrow prototype as complete.
